src/components/course/data/utils.jsx

```diff
@@ -149,83 +150,111 @@ export function findCouponCodeForCourse(couponCodes, catalogList = []) {
   }));
 }
 
+/**
+ * Determines the sort order of two enterprise offers based on the
+ * specified property for the user. Returns -1 if the first offer should
+ * be prioritized, 1 if the second offer should be prioritized, and 0 if
+ * there is no difference in priority.
+ *
+ * @param {object} args
+ * @param {object} args.firstOffer An enterprise offer to compare with the second offer.
+ * @param {object} args.secondOffer An enterprise offer to compare with the first offer.
+ *
+ * @returns The sort comparison value.
+ */
+export const compareOffersByProperty = ({ firstOffer, secondOffer, property }) => {
+  const firstOfferValue = firstOffer[property];
+  const secondOfferValue = secondOffer[property];
+  if (firstOfferValue && secondOfferValue) {
+    if (firstOfferValue < secondOfferValue) { return -1; }
+    if (firstOfferValue >= secondOfferValue) { return 1; }
+  }
+  if (!firstOfferValue && secondOfferValue) { return -1; }
+  if (firstOfferValue && !secondOfferValue) { return 1; }
+  return undefined;
+};
+
 /**
  * Returns an applicable enterprise offer to the specified enterprise catalogs, if one exists, with the
  * following prioritization:
  *   - Offer with no bookings limit (global or user)
+ *   - Offer with no applications limit (global or user)
  *   - Offer with user bookings limit
  *   - Offer with global bookings limit
+ *   - Offer with user enrollment limit
+ *   - Offer with global enrollment limit
  *
  * @param {array} enterpriseOffers List of enterprise offers available for the enterprise customer.
- * @param {array} catalogList List of enterprise catalog UUIDs associated with a given course.
  * @param {number} coursePrice The price of the course.
  *
  * @returns An object containing the metadata for the enterprise offer, if any, most applicable for
- * the specified enterporise catalog uuids and course price.
+ * the specified enterprise catalog uuids and course price.
  */
 export const findEnterpriseOfferForCourse = ({
```

Comment:
I would add additional testing to catch the new logic added in this function

Reply:
Yeah, I deferred on these tests because I plan on spending more time to verify/refactor the sorting logic per Marlon's earlier comments.